src/System.Buffers.ReaderWriter/System/Buffers/Writer/OutputWriter_overloads.cs

```diff
@@ -0,0 +1,114 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers.Text;
+using System.Runtime.CompilerServices;
+
+namespace System.Buffers.Writer
+{
+    public ref partial struct BufferWriter<T> where T : IBufferWriter<byte>
```

Comment:
Why does this need to be generic when the T is constrained to only a single type?

Reply:
So that IBufferWriter implementations can be structs and be written to without boxing.